From 60efaa9d93156556979f39a487ddc455dda11c88 Mon Sep 17 00:00:00 2001
From: Zhoujie Wu <zjwu@marvell.com>
Date: Tue, 11 Apr 2017 11:42:10 -0700
Subject: [PATCH 5/6] dts: armada-ap806: reserve 16M memory for tee

Reserve 16M memory for TEE, starts from 0x4400000.

These memory are used for TEE code, data, communication channel
and shared memory with Non-Trusted world.

For the TEE code and data section, it will be configured unacceable
to Non-Trusted World by memory controller.

The rest memories are either used for communication or shared memory
between Trusted-World and Non-trusted World, which is TEE implementation
defined and may be managed by the TEE driver.

Change-Id: Ifb324ec18e9732c66911706dedfced8c4ee8737a
Signed-off-by: Zhoujie Wu <zjwu@marvell.com>
---
 arch/arm64/boot/dts/marvell/armada-ap806.dtsi | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/arch/arm64/boot/dts/marvell/armada-ap806.dtsi b/arch/arm64/boot/dts/marvell/armada-ap806.dtsi
index 9c4a4b1..26c0a4d 100644
--- a/arch/arm64/boot/dts/marvell/armada-ap806.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-ap806.dtsi
@@ -73,6 +73,11 @@
 			reg = <0 0x4000000 0 0x200000>;
 			no-map;
 		};
+
+		tee@4400000 {
+			reg = <0 0x4400000 0 0x1000000>;
+			no-map;
+		};
 	};
 
 	/* CPUfreq Cluster settings:
-- 
2.7.4

